client/extensions/woocommerce/components/product-search/item.js

```diff
@@ -0,0 +1,90 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import CompactCard from 'components/card/compact';
+import { fetchProductVariations } from 'woocommerce/state/sites/product-variations/actions';
+import formattedVariationName from 'woocommerce/lib/formatted-variation-name';
+import getKeyboardHandler from 'woocommerce/lib/get-keyboard-handler';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import { getVariationsForProduct } from 'woocommerce/state/sites/product-variations/selectors';
+
+class ProductItem extends Component {
+	componentDidMount() {
+		const { siteId, productId } = this.props;
+
+		if ( siteId ) {
+			this.props.fetchProductVariations( siteId, productId );
+		}
+	}
+
+	componentWillReceiveProps( newProps ) {
+		const { siteId, productId } = this.props;
+		if ( newProps.productId !== productId || newProps.siteId !== siteId ) {
+			this.props.fetchProductVariations( newProps.siteId, newProps.productId );
+		}
+	}
+
+	onClick = () => {
+		const { product } = this.props;
+		this.props.onClick( product );
+	};
+
+	renderItem = product => {
+		const featuredImage = product.images && product.images[ 0 ];
```

Comment:
What do you think to using `_.get` here?
I put together a [quick demo](https://runkit.com/spen/59cc49e6a000c80012b0ac25) of how the case would be handled: